### PR TITLE
Deprecate 2x artifacts

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: agent/instrumentation/applicationinsights-web-2.3/src/main/java/io/opentelemetry/javaagent/instrumentation/applicationinsightsweb/RequestTelemetryContextInstrumentation.java [hunk 1]
```diff
@@ -115,24 +114,6 @@ public static void methodExit(
     }
   }
 
-  public static class TracestateBuilder implements BiConsumer<String, String> {
```
Comment (trask) [inline agent/instrumentation/applicationinsights-web-2.3/src/main/java/io/opentelemetry/javaagent/instrumentation/applicationinsightsweb/RequestTelemetryContextInstrumentation.java:118]: just moved this out into top-level class

File: legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/util/LocalStringsUtils.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZZ");
+
+  public static boolean isNullOrEmpty(String value) {
+    return value == null || value.isEmpty();
+  }
+
+  @SuppressFBWarnings(
+      value = "SECPR", // Predictable pseudorandom number generator
+      justification = "Predictable random is ok for telemetry id")
+  public static String generateRandomIntegerId() {
```
Comment (trask) [inline legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/util/LocalStringsUtils.java:41]: TODO (trask) is this method needed?

File: legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/util/Sanitizer.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+package com.microsoft.applicationinsights.internal.util;
+
+import java.net.URI;
+import javax.annotation.Nullable;
+
+public final class Sanitizer {
+
+  @Nullable
+  public static URI safeStringToUri(String url) {
```
Comment (trask) [inline legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/util/Sanitizer.java:30]: TODO (trask) inline into single caller

File: legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/StackFrame.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+package com.microsoft.applicationinsights.internal.schemav2;
+
+public class StackFrame {
+
+  private int level;
+  private String method;
+  private String assembly;
+  private String fileName;
+  private int line;
```
Comment (trask) [inline legacy-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/StackFrame.java:30]: TODO (trask) review all fields that are bridged and remove those that are not bridged

**Comment by trask** [comment]
TODO
* add new artifacts to release pipeline
* reconsider 3.0.0 as core and web artifact version
* copy LegacySdkWebInterop and run also against 3.x

**Review by heyams — APPROVED** [review]

**Pull request merged**